### PR TITLE
[SPARK-27489][WEBUI] UI updates to show executor resource information

### PR DESCRIPTION
**Pull request by tgravescs**
### What changes were proposed in this pull request?
We are adding other resource type support to the executors and Spark. We should show the resource information for each executor on the UI Executors page.
This also adds a toggle button to show the resources column.  It is off by default.

executorui1

Screenshot from 2019-08-28 14-56-26

### Why are the changes needed?
to show user what resources the executors have. Like Gpus, fpgas, etc

### Does this PR introduce any user-facing change?
Yes introduces UI and rest api changes to show the resources

### How was this patch tested?
Unit tests and manual UI tests on yarn and standalone modes.

### CODE REVIEW

**Review by felixcheung — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@gengliangwang FYI

**Comment by HyukjinKwon** [comment]
Seems fine to me too

**Comment by HyukjinKwon** [comment]
@gengliangwang FYI

File: core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala [hunk 1]
```diff
@@ -146,6 +150,7 @@ class JsonProtocolSuite extends SparkFunSuite {
   test("Dependent Classes") {
     val logUrlMap = Map("stderr" -> "mystderr", "stdout" -> "mystdout").toMap
     val attributes = Map("ContainerId" -> "ct1", "User" -> "spark").toMap
+    val resources = Map(ResourceUtils.GPU -> Array[String]("0")).toMap
```
Comment (gengliangwang) [inline core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala]: I think we can remove this line

**Review by gengliangwang — APPROVED** [review]
LGTM except one comment

File: core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala (hunk 1 above)
Comment (tgravescs) [reply core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala]: nice catched, removing

**Comment by tgravescs** [comment]
test this please

**Comment by tgravescs** [comment]
test this please

**Comment by gengliangwang** [comment]
Thanks, merge to master

**Pull request closed without merging**